docs/DeveloperGuide.md

```diff
@@ -2,36 +2,80 @@
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+Given below is an overview of the main components and how they interact with each other.
 
+### Data component
+The data component consists of customers, flight bookings, menu items, routes data as well as exceptions and validity checkers.<br>
+The actual data objects are encapsulated in the respective data managers. For example, the list of Customer objects are maintained in the CustomerManager object.<br>
+In this way, data are encapsulated and decoupled from higher layers.
+Manipulation of data can only be performed through data managers to ensure data integrity.
+
+### Command component
+Command class is an abstract class, which is then extended as specific commands such as BookCommand and AddRouteCommand.<br>
+Each command interacts with data managers to create, retrieve, delete or modify data.<br>
+This design makes use of polymorphism, such that duplication is avoided and the functionalities can be easily expanded.
+
+### Parser component
+Parser component consists of a general parser and dedicated parsers for customers data and flight bookings respectively. <br>
+The parser object takes in user input string, extracts desired information and returns a command object.<br>
+The parser class decouples the command class from parsing input. Thus, each class has only one dedicated responsibility.
+
+### Storage component
+Each data class has a file manager to handle data storage, such as CustomerFileManager for customer data and RouteFileManager for route data.<br>
+The file manager is responsible for decoding, encoding, saving and loading data from text file.
+
+### Common component
+Classes used by multiple components are in the `arcs.common` package.
+
+## Implementation
+This section describes how certain features are implemented. Only part of the features with noteworthy details are mentioned here.
+
+### 1. Bye command
+1.1 The following sequence diagram shows how the Bye Command works.
+![](ByeCommand.png)
 
 ## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+For airlines counter staff who are fast typist to easily manage flight bookings.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+Given that the pandemic is ending, more and more people are resuming air travel. <br>
+Therefore, airlines counter staff need a fast an efficient way to manage the many flight bookings.
 
 ## User Stories
 
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
-
+|v1.0|Counter Staff|Add in flight menu item|Add menu items available for in-flight meals|
```

Comment:
Besides indicating 'Counter Staff' as the user, you could explain the level of technical competency of the user e.g. First-time user, Experienced User etc.